patches/0003-Create-app-tray-icon.patch

```diff
@@ -0,0 +1,117 @@
+     electron_1.app.on('window-all-closed', () => {
+-        if ([platform_js_1.Platform.WINDOWS, platform_js_1.Platform.LINUX].includes(deviceInfo_js_1.devicePlatform)) {
++        if (!isTrayEnabled || isPlatformWindows) {
+             electron_1.app.quit();
+         }
+     });
+@@ -54,18 +56,10 @@ const handleWindowLifecycleEvents = (window) => {
+     window.on('close', (event) => {
+-        if (deviceInfo_js_1.devicePlatform !== platform_js_1.Platform.MACOS) {
+-            return;
+-        }
++        if (!isTrayEnabled) {
++            electron_1.app.quit();
++        }
+         if (state_js_1.state.willQuit) {
+             return;
+         }
+         event.preventDefault();
+-        if (window.isFullScreen()) {
+-            window.once('leave-full-screen', () => {
+-                (0, createWindow_js_1.toggleWindowVisibility)(window, false);
+-            });
+-            window.setFullScreen(false);
+-        }
+-        else {
+-            (0, createWindow_js_1.toggleWindowVisibility)(window, false);
+-        }
++        (0, createWindow_js_1.toggleWindowVisibility)(window, false);
+     });
+
+
+diff --git a/main/lib/tray.js b/main/lib/tray.js
+--- a/main/lib/tray.js
++++ b/main/lib/tray.js
+@@ -5,3 +5,2 @@ 
+ const node_path_1 = require("node:path");
+-const updater_js_1 = require("./updater.js");
+ const i18n_js_1 = require("./i18n.js");
+@@ -14,4 +13,6 @@ 
+ const trayIcon = () => {
+-    const iconPath = (0, node_path_1.join)(process.resourcesPath, 'assets', 'icon.ico');
++    const isTrayEnabled = process.env.TRAY_ENABLED;
++    const iconFile = isTrayEnabled == 2 ? 'favicon-dark.png' : (isTrayEnabled == 3 ? `favicon-light.png` : 'favicon.png');
++    const iconPath = (0, node_path_1.join)(__dirname, '../../app', iconFile);
+     return electron_1.nativeImage.createFromPath(iconPath);
+ };
+@@ -18,3 +20,2 @@ 
+ const createContextMenu = (window) => {
+-    const updater = (0, updater_js_1.getUpdater)();
+     const windowStateLabel = state_js_1.state.isWindowHidden || window.isMinimized()
+@@ -32,8 +33,2 @@ const createContextMenu = (window) => {
+         },
+-        {
+-            label: (0, i18n_js_1.formatMessage)({ id: 'desktop.check-for-updates' }),
+-            click() {
+-                updater.check();
+-            }
+-        },
+         {
+@@ -83,8 +79,6 @@ const setupTray = (window) => {
+     (0, exports.updateTrayMenu)(window);
+-    tray.on('click', () => {
+-        (0, createWindow_js_1.toggleWindowState)(window);
+-    });
+-    tray.on('double-click', () => {
+-        (0, createWindow_js_1.toggleWindowState)(window);
+-    });
++    tray.on('click', () => {
++        window.show();
++    });
++    tray.setIgnoreDoubleClickEvents(true);
+ };
+
+
+diff --git a/package.json b/package.json
+--- a/package.json
++++ b/package.json
+@@ -2,3 +2,3 @@ {
+     "private": true,
+-    "name": "YandexMusic",
++    "name": "yandex-music",
```

Comment:
According to the freedesktop standard, the .desktop file name must match the app_id. KDE follows this standard.